platform/darwin/scripts/generate-style-code.js

```diff
@@ -187,14 +187,22 @@ global.propertyType = function (property, _private) {
     return _private ? `id <MGLStyleAttributeValue, MGLStyleAttributeValue_Private>` : `id <MGLStyleAttributeValue>`;
 };
 
+global.initLayerIdentifierOnly = function (layerType) {
+    return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);`
+}
+
 global.initLayer = function (layerType) {
     if (layerType == "background") {
-        return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);`
+       return `_layer = new mbgl::style::${camelize(layerType)}Layer(layerIdentifier.UTF8String);` 
```

Comment:
We're leaking this layer and all other `mbgl::style::layers`. Should we delete it in `-dealloc` or rewrite the getters and setters to make them use the layer in a lazy way?